src/Particle/SoaDistanceTableAA.h

```diff
@@ -66,13 +66,13 @@ struct SoaDistanceTableAA : public DTD_BConds<T, D, SC>, public DistanceTableDat
     // initialize memory containers and views
     Ntargets_padded         = getAlignedSize<T>(N_targets);
     const size_t total_size = compute_size(N_targets);
-    memory_pool_displs_.resize(total_size * D);
+    memory_pool_.resize(total_size * (1 + D));
     distances_.resize(N_targets);
     displacements_.resize(N_targets);
     for (int i = 0; i < N_targets; ++i)
     {
-      distances_[i].resize(Ntargets_padded);
-      displacements_[i].attachReference(i, total_size, memory_pool_displs_.data() + compute_size(i));
+      distances_[i].attachReference(memory_pool_.data() + compute_size(i), i);
```

Comment:
Its hard to understand whats going on here. I tried defining similar functions to compute_size in the python shell.

lets say sizeof(T) is 8 and align is 32
computesize() mapped over (0, 1, 2, 3,4, 5) -> (0,4,8,12,16,24)
So I'm attaching references every 4 T's (actually RealTypes because thats what memory_pool_ is) but the size of distances_ grows larger than 4 T's. clearly starts to be bad.  Hopefully I just don't understand compute_size.  

Yet I just don't trust this code without tests.  It seems like the exact details of this memory layout are important but they're cryptic.

Reply:
row 0, size 0
row 1, size 4
row 2, size 4,
row 3, size 4
row 4, size 8
row 5, size 8

(0, 1, 2, 3, 4, 5 ) -> (0, 4, 8, 12, 20, 28)

Reply:
Unit test added.

Reply:
Ok I get it now, while it seems i did port it correctly I did fail to do some basic arithmetic in my head correctly 24-16 != 4.  The sizes make sense if the diagonal isn't included.